### DISCUSSION TITLE
The "wall" tag in the motion detector does not work as expected. It does not see the walls of the cave.

### DISCUSSION THREAD

**Discussion by pavel4win** [FakeFishGames/Barotrauma#14970] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

I tried to make a blocker for a depth charge launcher. So that the launch would be blocked if there is any creature or obstacle (wall) 15 meters below. So that the atomic bomb I dropped would not damage my submarine.
I found that the motion detector does not see the walls of the cave even if the "wall" tag is enabled.
The motion detector also does not seem to see the walls of the boat itself.
It turns out that this "wall" tag will only see artificial walls of other structures? Did it see the walls of stations, wrecks, enemy submarines, ruins?
IMHO this is not correct. How to make a blocker if there is a seabed under the boat and it is dangerous to drop a bomb?
The wall tag, IMHO, should cover absolutely any obstacles.
Including piezoelectric crystals, mineral clusters, and other similar things.
Probably, except for living creatures, since there are separate tags for them. But personally I would make a separate tag "everything" that covers any obstacles that a collision can occur with. Creatures, walls, etc. not counting the detector itself, of course.

### Reproduction steps

1. Start game round with motion detector and electrical glasses in inventory
2. Attach motion detector near bottom your sub.
3. set motion detector tags to "wall" only, MD cover area to 1000 x and 1000 y
4. descent to seabed
5. Look at motion detector output.
6. It is remain zero despite seabed under the sub

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Happens in both single player and multiplayer

### Version

v1.6.17.0 (Unto the Breach Update)

### Which operating system did you encounter this bug on?

Windows

**Comment by Regalis11** [comment]
Thank you for the report!

I unfortunately could not reproduce the issue, the sensor seems to be able to detect level walls.

https://github.com/user-attachments/assets/5d6aef81-f263-4635-8ed1-dad2bf849020

It's worth noting though that the sensor needs to be outside, it cannot "see" the level walls through the submarine's walls even if the trigger area is large enough to extend outside the sub.

**Reply by pavel4win** [reply]
So I misunderstood how the "wall" tag works.
My apologies. I'm closing the discussion, if I find anything wrong, I'll open it again.

**Reply by Remodor** [reply]
Please show me how you do this for the Azimuth in a single player campaign. It doesn't work because you cannot place this "outside" the sub without being inside the editor.
